osquery/tables/system/darwin/user_groups.mm

```diff
@@ -151,24 +160,28 @@ QueryData genUsers(QueryContext& context) {
         genODEntries(kODRecordTypeUsers, username, usernames);
 
         Row r;
-        r["username"] = pwd->pw_name;
-        r["is_hidden"] = INTEGER(usernames[r["username"]]);
         genUserRow(r, pwd);
+        r["is_hidden"] = INTEGER(usernames[r["username"]]);
         results.push_back(r);
       }
     } else {
       std::map<std::string, bool> usernames;
       genODEntries(kODRecordTypeUsers, nil, usernames);
       for (const auto& username : usernames) {
+        // opendirectory and getpwnam are documented as having
+        // different code paths. Thus we may see cases where
+        // genODEntries produces responses that are not in
+        // getpwnam. So with a surfeit of caution we populate some of
+        // the row here
+        Row r;
+        r["username"] = TEXT(username.first.c_str());
+        r["is_hidden"] = INTEGER(username.second);
+
         struct passwd* pwd = getpwnam(username.first.c_str());
-        if (pwd == nullptr) {
-          continue;
+        if (pwd != nullptr) {
```

Comment:
For example, ditch line 177 and write this here as:
```cpp
if (pwd != nullptr) {
  genUserRow(r, pwd);
} else {
  r["username"] = TEXT(username.first.c_str());
}
```

Reply:
Now it's DRY ;)

Reply:
Okay. It all feels kinda grotty either way really. Implemented